src/irmin-containers/containers.mli

```diff
@@ -0,0 +1,68 @@
+(*
+ * Copyright (c) 2020 KC Sivaramakrishnan <kc@kcsrk.info>
+ * Copyright (c) 2020 Anirudh Sunder Raj <anirudh6626@gmail.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+module type S = sig
+  (** Signature which all containers must satisfy *)
+
+  type t
+  (** Type for stores *)
+
+  type repo
+  (** Type for repositories *)
+
+  type branch
+  (** Type for persistent branch identifiers *)
+
+  type key
+  (** Type for keys *)
+
+  type contents
+  (** Type for contents *)
+
+  module Store : Irmin.S
```

Comment:
This is an odd inclusion in the signature. The `Irmin.S` types aren't constrained to be equal to the top-level ones, and `Irmin.S` contains very similar operations. What's the idea here?

Reply:
This was done to enable the user to use all the `Irmin.S` functions, while some frequently used functions were included directly. I have removed the included functions, so that now all the cloning, branching, etc must be done through `Store`.